CMakeLists.txt

```diff
@@ -56,297 +63,80 @@ zephyr_include_directories(
-# ==========================================================================
-if(W MATCHES "1")
-  zephyr_compile_options(
-    -Wextra
-    -Wunused
-    -Wno-unused-parameter
-    -Wmissing-declarations
-    -Wmissing-format-attribute
-    -Wold-style-definition
-    )
-  zephyr_cc_option(
-    -Wmissing-prototypes
-    -Wmissing-include-dirs
-    -Wunused-but-set-variable
-    -Wno-missing-field-initializers
-    )
-endif()
-
-if(W MATCHES "2")
-  zephyr_compile_options(
-    -Waggregate-return
-    -Wcast-align
-    -Wdisabled-optimization
-    -Wnested-externs
-    -Wshadow
-    )
-  zephyr_cc_option(
-    -Wlogical-op
-    -Wmissing-field-initializers
-    )
-endif()
-
-if(W MATCHES "3")
-  zephyr_compile_options(
-    -Wbad-function-cast
-    -Wcast-qual
-    -Wconversion
-    -Wpacked
-    -Wpadded
-    -Wpointer-arith
-    -Wredundant-decls
-    -Wswitch-default
-    )
-  zephyr_cc_option(
-    -Wpacked-bitfield-compat
-    -Wvla
-    )
-endif()
+# Discover if the default C standard has been overridden
+# So far, only POSIX-based boards does this
+get_property(CSTD GLOBAL PROPERTY CSTD)
+set_ifndef(CSTD c99)  # Set the default C standard to ISO/IEC 9899:1999
+
+
+## @Brief: toolchain_cc: Macro that expands into compiler options.
+## @Brief: toolchain_ld: Macro that expands into linker options.
+## @Details:
+##   toolchain_cc must have been defined by compiler/${COMPILER}.cmake, where
+##     COMPILER was set by toolchain/${ZEPHYR_TOOLCHAIN_VARIANT}.cmake.
+##   toolchain_ld must have been defined by linker/${LINKER}.cmake, where
+##     LINKER was set by toolchain/${ZEPHYR_TOOLCHAIN_VARIANT}.cmake.
+##
+##   We can classify compiler/linker options into 4 groups:
+##
+##     1. Toolchain-specific and Zephyr-agnostic:
+##          Covers: ABI, call-convention, code generation options, etc.
+##          Where: Defines and includes should be should be captured by
+##            {TOOLCHAIN_C_FLAGS, TOOLCHAIN_INCLUDES} set in ${COMPILER}.cmake.
+##            Compiler options that are not defines or includes, should be set
+##            directly in ${COMPILER}.cmake or ${LINKER}.cmake.
+##
+##  --> 2. Toolchain-specific and Zephyr-specific:
+##          Covers:
+##            a. Mapping Kconfig "intents" to toolchain-specific flags (e.g. W,
+##              CONFIG_SPEED_OPTIMIZATIONS, CONFIG_READABLE_ASM).
+##            b. Mapping Zephyr expectations to toolchain-specific flags
+##              PIC/PIE, freestanding,  inclusion of autoconf.h, expected
+##              overflow behavior, etc.
+##          Where: Should be placed in toolchain_cc() or toolchain_ld().
+##          Note: It may be neat, but optional, to further subdivide
+##            toolchain_cc down into toolchain_cc_{security,optimizations,cpp,
+##            freestanding,warnings,diagnostics,asm}. Subdividing like this can
+##            improve reuse (e.g. Clang accepts the gcc-names of C++ standards),
+##            and demarcate responsibility.
```

Comment:
"demarcate"

It is a part of the Zephyr coding standard to use simple english.

Reply:
OK, will fix.